File: src/main/java/org/example/expert/domain/manager/service/LogService.java
package org.example.expert.domain.manager.service;

import org.example.expert.domain.manager.entity.Log;
import org.example.expert.domain.manager.repository.LogRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Propagation;
import org.springframework.transaction.annotation.Transactional;

@Service
public class LogService {

	private final LogRepository logRepository;

	public LogService(LogRepository logRepository) {
		this.logRepository = logRepository;
	}

	@Transactional(propagation = Propagation.REQUIRES_NEW)
	public void saveLog(String action, String details) {
		Log log = new Log();
		log.setAction(action);
		log.setDetails(details);
		logRepository.save(log);
	}
}

File: src/main/kotlin/com/example/domain/manager/service/LogService.kt
package com.example.domain.manager.service

import com.example.domain.manager.entity.Log
import com.example.domain.manager.repository.LogRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Propagation
import org.springframework.transaction.annotation.Transactional

@Service
class LogService(

    private val logRepository: LogRepository
) {

    @Transactional(propagation = Propagation.REQUIRES_NEW)
    fun saveLog(action: String, details: String) {
        val log = Log(action = action, details = details)
        logRepository.save(log)
    }
}

